src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Extensions/StartupExtensions.cs

```diff
@@ -17,12 +18,45 @@
 using Microsoft.EntityFrameworkCore;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
 using Microsoft.OpenApi.Models;
 
 namespace DevAdventCalendarCompetition.Extensions
 {
     public static class StartupExtensions
     {
+        public static IServiceCollection AddAdventConfiguration(this IServiceCollection services, IConfiguration configuration)
+        {
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
+
+            services.Configure<AdventSettings>(a =>
+                {
+                    string defaultDateTimeFormat = "dd-MM-yyyy";
+                    var isAdventEndDate = configuration.GetSection("Competition:EndDate").Value;
```

Comment:
I suggest using _adventStart/EndDate_, because _IsAdventStart/EndDate_ suggests that value is boolean